subsys/settings/src/settings_zms.c

```diff
@@ -445,6 +445,9 @@ static int settings_zms_save(struct settings_store *cs, const char *name, const
 		/* Settings entry exist, let's verify if this is the same
 		 * name
 		 */
+		if (rc >= sizeof(rdname)) {
+			return -EOVERFLOW;
+		}
```

Comment:
You might not need this anymore, as `zms_read()` takes care of always have some room to add `\0`, so `rc` will always be < `sizeof(rdname)`... However, it might be possible to turn it into an assert:

```suggestion
		__ASSERT_NO_MSG(rc < sizeof(rdname));
```